Lib/test/test_structseq.py

```diff
@@ -108,7 +110,101 @@ def __len__(self):
 
     def test_reduce(self):
         t = time.gmtime()
-        x = t.__reduce__()
+        cls, (tup, dct) = t.__reduce__()
+        self.assertIs(cls, time.struct_time)
+        self.assertIsInstance(tup, tuple)
+        self.assertIsInstance(dct, dict)
+        self.assertEqual(tup, t)
+        self.assertEqual(len(tup), len(t))
+        self.assertEqual(len(dct), t.n_fields - t.n_sequence_fields)
+
+    def test_reduce_with_unnamed_fields(self):
+        assert os.stat_result.n_unnamed_fields > 0
+
+        r = os.stat_result(range(os.stat_result.n_sequence_fields),
+                           {'st_atime': 1.0, 'st_atime_ns': 2.0})
+        self.assertEqual(r.st_atime, 1.0)
+        cls, (tup, dct) = r.__reduce__()
+        self.assertIs(cls, os.stat_result)
+        self.assertIsInstance(tup, tuple)
+        self.assertIsInstance(dct, dict)
+        self.assertEqual(tup, tuple(r))
+        self.assertIn('st_atime', dct)
+        self.assertIn('st_atime_ns', dct)
+        self.assertEqual(len(dct), r.n_fields - r.n_sequence_fields)
+
+    def test_pickling(self):
+        t = time.gmtime()
+        for proto in range(pickle.HIGHEST_PROTOCOL + 1):
+            p = pickle.dumps(t, proto)
+            t2 = pickle.loads(p)
+            self.assertEqual(t.__class__, t2.__class__)
```

Comment:
Nitpick: usually the left argument of `assertEqual()` is the actual value, and the right argument is the expected value. It is not always consistent in old tests, and it does not affect the correctness of tests, but it is easier to read if it is in such order.

In this case it should be `self.assertEqual(t2.__class__, t.__class__)` (and similarly in other tests).